dao/src/main/kotlin/repositories/DaoSecretRepository.kt

```diff
@@ -43,23 +44,17 @@ import org.slf4j.LoggerFactory
 private val logger = LoggerFactory.getLogger(DaoSecretRepository::class.java)
 
 class DaoSecretRepository(private val db: Database) : SecretRepository {
-    override fun create(
-        path: String,
-        name: String,
-        description: String?,
-        organizationId: Long?,
-        productId: Long?,
-        repositoryId: Long?
-    ) = db.blockingQuery {
-        SecretDao.new {
-            this.path = path
-            this.name = name
-            this.description = description
-            this.organization = organizationId?.let { OrganizationDao[it] }
-            this.product = productId?.let { ProductDao[it] }
-            this.repository = repositoryId?.let { RepositoryDao[it] }
-        }.mapToModel()
-    }
+    override fun create(path: String, name: String, description: String?, entity: Entity, id: Long) =
+        db.blockingQuery {
+            SecretDao.new {
+                this.path = path
+                this.name = name
+                this.description = description
+                this.organization = id.takeIf { entity == Entity.ORGANIZATION }?.let { OrganizationDao[it] }
+                this.product = id.takeIf { entity == Entity.PRODUCT }?.let { ProductDao[it] }
+                this.repository = id.takeIf { entity == Entity.REPOSITORY }?.let { RepositoryDao[it] }
```

Comment:
`if` would be shorter and to me also more readable:

```kotlin
this.organization = if (entity == Entity.ORGANIZATION) OrganizationDao[it] else null
```

But it should also work with `when`:
```kotlin
when (entity) {
  Entity.ORGANIZATION -> this.organization = OrganizationDao[id]
  ...
}
```

Reply:
What I usually like about the `takeIf` approach is that the thing to assign from is listed first, instead of last / in the middle with the `if` approach.

Anyway, if `when` works here as entities are `null` by default, that's yet nicer. I'll use that.

Reply:
> What I usually like about the `takeIf` approach is that the thing to assign from is listed first, instead of last / in the middle with the `if` approach.

It's a matter of taste and being familiar with the pattern, but to me it's actually the opposite, close to an anti-pattern, especially if the condition does not refer to the thing `takeIf` is called on, like in this case. The if condition I can read from left to right like human language and understand on the first read what happens: "If the condition is fulfilled do this, otherwise that."
With `takeIf` in combination with `let` I always need additional thinking steps, I have to jump back to the beginning after understanding the condition and then see how the thing `takeIf` was called on is used in the `let` block, and remember that if the condition is false the whole expression evaluates to null.

> Anyway, if `when` works here as entities are `null` by default, that's yet nicer. I'll use that.

I only hope it works but I'm not sure. I guess you'll find out.

Reply:
> I guess you'll find out.

I hope the exitsing test coverage does for me 😉

Reply:
Just a point, for someone not too familiar with Kotlin, of the options above, the one below is by far the simplest to quickly comprehend for me. 

> ```kotlin
> this.organization = if (entity == Entity.ORGANIZATION) OrganizationDao[it] else null
> ```